mp/src/game/shared/momentum/run/mom_replay_factory.cpp

```diff
@@ -98,4 +120,159 @@ CMomReplayBase* CMomReplayFactory::LoadReplayFile(const char* pFileName, bool bF
     return toReturn;
 }
 
+#ifdef CLIENT_DLL
+LocalTimesStatus_t CMomReplayFactory::AsyncLoadReplayFileMultiple(const CUtlStringList& filenames, const char *pathID,
+                                                                 MomLocalReplayCallback_t callback,
+                                                                 MomReplayAsyncHandle* outHandle)
+{
+    auto *internalRequest = new MomReplayLoadRequestInternal;
+    if (outHandle)
+        *outHandle = internalRequest->handle;
+    requests.AddToTail(internalRequest);
+    internalRequest->callback = callback;
+
+    size_t count = filenames.Size();
+
+    auto fsControls = new FSAsyncControl_t[count];
+    auto fsRequests = new FileAsyncRequest_t[count]; // they SHOULD get deleted by filesystem
+
+    internalRequest->arrControls = fsControls;
+    internalRequest->controls_count = count;
+    internalRequest->controls_pending = count;
+
+    FOR_EACH_VEC (filenames, i)
```

Comment:
Since this one's a macro you won't want spacing between the macro name and the params.